// @aws-sdk/client-cognito-identity-provider/dist-es/models/models_1.js
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { CognitoIdentityProviderServiceException as __BaseException } from "./CognitoIdentityProviderServiceException";
import { AttributeTypeFilterSensitiveLog, UserPoolClientTypeFilterSensitiveLog, } from "./models_0";
export class EnableSoftwareTokenMFAException extends __BaseException {
    constructor(opts) {
        super({
            name: "EnableSoftwareTokenMFAException",
            $fault: "client",
            ...opts,
        });
        this.name = "EnableSoftwareTokenMFAException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EnableSoftwareTokenMFAException.prototype);
    }
}
export var VerifySoftwareTokenResponseType;
(function (VerifySoftwareTokenResponseType) {
    VerifySoftwareTokenResponseType["ERROR"] = "ERROR";
    VerifySoftwareTokenResponseType["SUCCESS"] = "SUCCESS";
})(VerifySoftwareTokenResponseType || (VerifySoftwareTokenResponseType = {}));
export const SignUpResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StartUserImportJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StartUserImportJobResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StopUserImportJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StopUserImportJobResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UntagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UntagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateAuthEventFeedbackRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Username && { Username: SENSITIVE_STRING }),
    ...(obj.FeedbackToken && { FeedbackToken: SENSITIVE_STRING }),
});
export const UpdateAuthEventFeedbackResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateDeviceStatusRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AccessToken && { AccessToken: SENSITIVE_STRING }),
});
export const UpdateDeviceStatusResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateGroupRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateGroupResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateIdentityProviderRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateIdentityProviderResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateResourceServerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateResourceServerResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateUserAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.UserAttributes && {
        UserAttributes: obj.UserAttributes.map((item) => AttributeTypeFilterSensitiveLog(item)),
    }),
    ...(obj.AccessToken && { AccessToken: SENSITIVE_STRING }),
});
export const UpdateUserAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateUserPoolRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateUserPoolResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateUserPoolClientRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ClientId && { ClientId: SENSITIVE_STRING }),
});
export const UpdateUserPoolClientResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.UserPoolClient && { UserPoolClient: UserPoolClientTypeFilterSensitiveLog(obj.UserPoolClient) }),
});
export const UpdateUserPoolDomainRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateUserPoolDomainResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const VerifySoftwareTokenRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AccessToken && { AccessToken: SENSITIVE_STRING }),
});
export const VerifySoftwareTokenResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const VerifyUserAttributeRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AccessToken && { AccessToken: SENSITIVE_STRING }),
});
export const VerifyUserAttributeResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
