// @aws-sdk/client-cognito-identity-provider/dist-es/commands/UpdateUserAttributesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateUserAttributesRequestFilterSensitiveLog, UpdateUserAttributesResponseFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_json1_1UpdateUserAttributesCommand, serializeAws_json1_1UpdateUserAttributesCommand, } from "../protocols/Aws_json1_1";
export class UpdateUserAttributesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CognitoIdentityProviderClient";
        const commandName = "UpdateUserAttributesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: UpdateUserAttributesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateUserAttributesResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1UpdateUserAttributesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1UpdateUserAttributesCommand(output, context);
    }
}
